db/in_memory_stats_history.h

```diff
@@ -26,7 +36,13 @@ class InMemoryStatsHistoryIterator final : public StatsHistoryIterator {
   bool Valid() const override;
   Status status() const override;
 
+  // Move to the next stats snapshot currently visible in
+  // DBImpl::stats_history_. Because of garbage collection, the next stats
+  // snapshot may or may not be right after the current one. When reading
+  // from DBImpl::stats_history_, this call will be protected by DB Mutex so it
+  // will not return partial or corrupted results.
```

Comment:
Does the comment belong to declaration, or definition? It does feel to me that few people will try to learn this information here.